src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -530,5 +535,115 @@ public static string[] GetLogicalDrives()
         {
             return DriveInfoInternal.GetLogicalDrives();
         }
+
+        /// <summary>Gets the path of the target of the specified link.</summary>
+        /// <param name="linkPath">A path to a link file.</param>
+        /// <param name="isDirectory">Whether the link represents a directory or not. Irrelevant in Unix since readlink does not care about the underlying type.</param>
+        /// <returns>If linkPath represents a link file and it exists, returns the link's target path.
+        /// If linkPath is not a link or the target does not exist, returns null.</returns>
+        internal static string? GetLinkTarget(ReadOnlySpan<char> linkPath, bool isDirectory) => Interop.Sys.ReadLink(linkPath);
+
+        /// <summary>
+        /// Creates a file symbolic link identified by path that points to pathToTarget.
+        /// </summary>
+        /// <param name="path">The path where the symbolic link should be created.</param>
+        /// <param name="pathToTarget">The path of the target to which the symbolic link points.</param>
+        /// <param name="isDirectory">True if the pathToTarget represents a directory or a symlink to a directory.</param>
+        internal static void CreateSymbolicLink(string path, string pathToTarget, bool isDirectory)
+        {
+            string pathToTargetFullPath = PathInternal.IsPartiallyQualified(pathToTarget.AsSpan()) ?
+                Path.Join(Path.GetDirectoryName(path), pathToTarget) : pathToTarget;
+
+            // Fail if the target exists but is not consistent with the expected filesystem entry type
+            if (Interop.Sys.LStat(pathToTargetFullPath, out Interop.Sys.FileStatus targetInfo) == 0)
```

Comment:
It might fail when `pathToTargetFullPath` doesn't exists, but that should not prevent us from finishing successfully since a link can be created pointing to a non-existent path.